### PR TITLE
Bug1133095 convert issue details to single models

### PR DESCRIPTION
**Pull request by schalkneethling**
This PR is getting to be to big so, I have split two items out into separate bugs so we can get this reviewed and merged. The new bugs are:

https://bugzilla.mozilla.org/show_bug.cgi?id=1146306
https://bugzilla.mozilla.org/show_bug.cgi?id=1146310

With that, let the review and feedback begin ;)

@ckprice @pmclanahan r?

### CODE REVIEW

**Comment by schalkneethling** [comment]
Also, split the actual code changes from the code cleanup so it is in two separate commits.

**Comment by schalkneethling** [comment]
Another point to note, the infographics are not shown in the new modal. This is intentional, as I want to add them back, one by one, to ensure we only add the one's we actually need, to ensure it uses the latest data provided by Greg and also, to make any changes/updates if there are bugs connected to the 'graphic'

**Comment by ckprice** [comment]
@schalkneethling thanks!

1. Any luck on removing the background image circle on all views except the Ecosystem? Can we get that fixed in this PR?
2. When I'm on 'Vitals' and I click on an issue, then close that issue, I should stay on the 'Vitals' view. Instead I am taken to the Ecosystem view. Same goes with the Checklist view

**Comment by schalkneethling** [comment]
@ckprice 1. I will be looking into that 2. hmmm, thought I opened a separate bug for that... Will open one.

I do not want to put more into this pull request and would instead have follow on pull requests for those items.

**Comment by ckprice** [comment]
@schalkneethling 

I saw [bug 1146306](https://bugzilla.mozilla.org/show_bug.cgi?id=1146306) after I posted the comment. I am noticing this outside of direct referrals, it also happens during normal browsing.

> I do not want to put more into this pull request and would instead have follow on pull requests for those items.

Okay.

**Comment by ckprice** [comment]
I opened [bug 1146404](https://bugzilla.mozilla.org/show_bug.cgi?id=1146404) for the background circle item.

**Comment by ckprice** [comment]
r+ on functionality. The main purpose of this PR was to change the structure of the modal issue presentation, which was accomplished. Separate bugs have been opened to address the issues we spoke to in our checkin.

@pmclanahan, @alexgibson r code?

**Comment by ckprice** [comment]
@schalkneethling just a note that it looks like our responsive layout got a bit hammered here.

**Comment by schalkneethling** [comment]
@ckprice yeah, I still need to figure out all the crazy CSS that is involved here so not surprise. I also want to test this on mobile because at the moment, we are including jQuery Mobile, and I am not sure why (even on desktop).

**Comment by schalkneethling** [comment]
There is also this link:
https://github.com/mozilla/lorax/blob/master/src/index.html#L28

That I assume is shown for mobile devices, although I cannot be sure but, once shown, it will have links to the about and legend modal, which we do not want, so that is something I will look into.

**Comment by alexgibson** [comment]
@schalkneethling is this PR ready for review, or would you like to work on the redponsive/mobile issues mentioned above first?

**Comment by schalkneethling** [comment]
@alexgibson this is good to go, I do not want to push more things into this PR so, we have split out the issues that has been identified into separate bugs. Thanks!

**Comment by alexgibson** [comment]
After opening an issue/detail, pressing the browser back button no longer returns to the main page

**Comment by alexgibson** [comment]
Related to this, if I'm on vitals or checklist view and open an issue, closing the modal (via the `x` button) returns me to ecosystem, and not the mode I was viewing.

**Comment by alexgibson** [comment]
Likely not related to this change, but if I switch modes too fast I get overlapping visuals:

fast-switch-modes

**Comment by alexgibson** [comment]
Escaping the modal is not currently possible via keyboard (would be nice to hit `esc` maybe, or at least tab to the close button).

**Comment by alexgibson** [comment]
Again likely unrelated to this change, but I find it difficult/impossible to open the "About" modal via the button in the bottom left corner?

File: src/app/lorax/directives/issue-all.js [hunk 1]
```diff
@@ -82,175 +74,23 @@ define(['jquery', 'webfontloader'], function ($, WebFont) {
...
-    };
+    IssueAllCtrl.prototype.open = function (issue, topic) {
+        this.issue = issue;
 
-    IssueAllCtrl.prototype.close = function () {
-        $(window).off('scroll', this._onScrollBind);
+        var issueModel = this._$scope.detail.model.getIssueById(this.issue);
+        this._windowService.setIssueMode(true);
+        issueModel._isVisible = true;
+        console.log(issueModel);
```
Comment (alexgibson) [inline src/app/lorax/directives/issue-all.js]: Need to remove this `console.log()`?

File: src/app/lorax/directives/modal-issue.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+        this._$scope = $scope;
+        this._$location = $location;
+        this._$timeout = $timeout;
+        this._windowService = windowService;
+        this._dataService = dataService;
+
+        $scope.modalIssue = {
+            open: false,
+            closeModal: this.closeModal.bind(this),
```
Comment (alexgibson) [inline src/app/lorax/directives/modal-issue.js]: Nit: remove trailing comma

File: src/app/lorax/directives/modal-issue.tpl.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<article class="modal modal--fade-show modal-issue" ng-show="modalIssue.open" >
+  <a href="#" class="modal__close modal__close-absolute icon-close" data-lorax-prevent-default ng-click="modalIssue.closeModal()">Close</a>
```
Comment (alexgibson) [inline src/app/lorax/directives/modal-issue.tpl.html]: Semantically would this be better suited as a `<button>`?

File: src/app/lorax/directives/modal-issue.tpl.html [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  <div class="container">
+    <h1>{{modalIssue.issue.getName()}}</h1>
```
Comment (alexgibson) [inline src/app/lorax/directives/modal-issue.tpl.html]: Are multiple `<h1>`'s still an accessibility no-go?

File: src/app/lorax/services/experience.js [hunk 5]
```diff
@@ -98,10 +88,8 @@ define(['jquery', 'experience/experience'], function ($, Experience) {
                 case 'checklist':
                     this._experience.showIssues();
                     break;
-                case 'tag':
-                    this._experience.showTagIssues(data);
-                    break;
                 case 'detail':
+                    console.log('case: detail calls');
```
Comment (alexgibson) [inline src/app/lorax/services/experience.js]: Remove `console.log()`?

File: src/app/lorax/services/experience.js [hunk 6]
```diff
@@ -121,9 +109,15 @@ define(['jquery', 'experience/experience'], function ($, Experience) {
         this._offset = offset;
     };
 
+    /**
+     * This function is called by _openIssue in src/scripts/experience/experience.js
```
Comment (alexgibson) [inline src/app/lorax/services/experience.js]: Love seeing useful comments like this :+1:

**Comment by alexgibson** [comment]
A few UX quirks and odd bugs to iron out (not necessarily related to this change), but overall this change looks good so far. FWIW, I find this new behavior a million times more intuitive than the endless scrolling of the old details view, in which I instantly felt lost and overwhelmed. :+1:

**Comment by alexgibson** [comment]
Also noticed if I can on a detail URL, it does not show and instead loads the experience view.

File: src/app/lorax/directives/modal-issue.tpl.html (hunk 3 above)
Comment (schalkneethling) [inline src/app/lorax/directives/modal-issue.tpl.html]: I agree, however, this is used as it is now in a few places so, I did not want to change that at the moment but, again, definitely something to rethink, and if for some reason that poses a significant issue, will at least add role=“button”

File: src/app/lorax/directives/modal-issue.tpl.html (hunk 4 above)
Comment (schalkneethling) [inline src/app/lorax/directives/modal-issue.tpl.html]: Yeah, I was wondering about that as well, I would think that one should not have more than one, to assist with document structure but, with the changes in HTML5 on can technically have more than one now, especially in different `<section>` and `<article>` blocks but, still not sure about this one. I will have a look at the overall document structure, which is a problem I identified before, and see how this impacts it. For now, I left it as it was.

**Comment by schalkneethling** [comment]
Responding from first to last comment and inline for comments on code bits:

1. Yes, this is a known issue and will be resolved in https://bugzilla.mozilla.org/show_bug.cgi?id=1146306
2. Known issue, will be resolved in a follow on bug
3. Yeah, performance, performance, performance ;) Lots more to do there
4. Yup, will be fixed as part of https://bugzilla.mozilla.org/show_bug.cgi?id=1146310
5. So you are the second person to mention that but, for some unknown reason, I cannot replicate it :-/ - Will look into this however and resolve in a follow on PR
6. Yes, this will be resolved in https://bugzilla.mozilla.org/show_bug.cgi?id=1146306

Thanks @alexgibson

File: src/app/lorax/directives/modal-issue.tpl.html (hunk 3 above)
Comment (alexgibson) [inline src/app/lorax/directives/modal-issue.tpl.html]: Understood, fine to follow up later as this is a relatively minor issue :+1:

File: src/app/lorax/directives/modal-issue.tpl.html (hunk 4 above)
Comment (alexgibson) [inline src/app/lorax/directives/modal-issue.tpl.html]: AFAIK, screen-readers have yet to implement the new HTML5 document outline algorithm, so while multiple `<h1>`'s are valid per the official spec, they are still somewhat confusing for AT users.

Fine to leave this for now though, can always follow up when you have a look at the overall structure :+1:

**Comment by alexgibson** [comment]
Looks like most of the outstanding bugs are covered in the followup bugs, and nits are addressed here so r+ from me.

As for (5.), it feels somewhat like a redraw issue to me. I'm on a retina screen if that helps at all. Happy to help test when you come to address this bug.

**Pull request merged**